Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonProducer.cc

```diff
@@ -281,7 +281,8 @@ void AlCaHcalHBHEMuonProducer::produce(edm::Event& iEvent, const edm::EventSetup
   const EcalSeverityLevelAlgo* sevlv = &iSetup.getData(tok_sevlv_);
   const CaloTopology* caloTopology = &iSetup.getData(tok_topo_);
   const HcalDbService* conditions = &iSetup.getData(tok_dbservice_);
-  HcalRespCorrs* respCorrs = new HcalRespCorrs(*resp);
+  HcalRespCorrs respCorrsObj(*resp);
+  HcalRespCorrs* respCorrs = &respCorrsObj;
   respCorrs->setTopo(theHBHETopology);
```

Comment:
I'm puzzled by this pattern of copying an EventSetup product (`resp`) and then setting a member variable of the copy from another EventSetup product (`theHBHETopology`). Wouldn't it be better to have the `HcalRespCorrs` in EventSetup initialized with the topology object (or something similar)?

Reply:
@bsunanda can you please comment?